### DISCUSSION TITLE
Impossible connection after changing NFS_RW mode for Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by ThierryM** [ltsp/ltsp#329] [Q&A]
Hi, 
Thanks to you, I 've succeeded in launching a Raspberry Pi 4 client from an Ubuntu Server 20.04 and LTSP 20 with an educationnal Raspberry Pi OS based distribution (Primtux4).
I detailled how I made here : https://lofurol.fr/joomla/electronique/raspberry-pi/263-utiliser-des-raspberry-pi-4-en-tant-que-clients-lourds
In the [documentation page](https://ltsp.org/docs/installation/raspbian/), I'm stuck at the "LTSP mode netbooting" step where it's impossible to launch the Raspberry Pi. All works fine, NFS_RW netbooting too, but the final step.
For information, the Primtux4 Image display at the starting a connexion screen with 4 users profiles (3 with no password and 1 with an administrator password).
I think I forget something about the /home clients but I don't find what...
Thanks for your help, regards,

Thierry

**Comment by alkisg** [comment]
If I understand you correctly, you're seeing the login manager screen (lightdm) and you try to login but you get back to the login screen?

If so, press Ctrl+Alt+F2 to switch to vt2, then try to login there. See what error messages are displayed there in the console.

Did you `apt install --install-recommends ltsp` in the chroot? That also installs the sshfs package, which is necessary for /home.

**Comment by ThierryM** [comment]
Thanks for your help again... and probably, not the last... ;-)
I only see the login manager when I'm in NFS_RW mode. In this mode, all it's working well.
The problem appears after I made [the last step](https://ltsp.org/docs/installation/raspbian/#ltsp-mode-netbooting). Then impossible to boot the Raspberry Pi 4, the boot stops and no GUI.
Yes I've done `apt install --install-recommends ltsp` in the Raspberry Pi 4 when I was in NFS_RW mode (before the last step).

I tried to see logs with the netconsole but nothing is displayed with this command : `socat UDP-LISTEN:6666,fork -`
I've tried too the commands on the server : 
`cp /usr/bin/sshfs /etc/ltsp/sshfs-$(uname -m)`
`ltsp initrd`
And it was a sshfs file for AMD 64 (the same as the server). But no effect.
Regards.

**Comment by alkisg** [comment]
Did you test Ctrl+Alt+F2?
Can you upload a video of the client boot process?

**Reply by ThierryM** [reply]
Yes, I tested Ctrl+Alt+F2 but nothing happened. I'm blocked. The only thing that I can do : it's unplugging the outlet ;-).
I try to to ulpoad a video tomorrow.

**Reply by ThierryM** [reply]
Hi,
You can see the video with the blocked state (LTSP Netbooting) here : https://tube-montpellier.beta.education.fr/videos/watch/39de0225-30d5-4745-a08e-d080f06f8d5a
See the attached file to see the screen at this state
![Blocked state](https://user-images.githubusercontent.com/1470067/99637198-cbc8a880-2a44-11eb-8a53-f412913c9367.jpg)

I posted too the booting in NFS_RW mode of the Raspberry Pi 4 and where all is fine : https://tube-montpellier.beta.education.fr/videos/watch/9d1bf94c-ed4e-4468-b628-394e2b4e6004

**Comment by alkisg** [comment]
Hi, in the https://ltsp.org/docs/installation/raspbian/#ltsp-mode-netbooting step, try to remove "quiet splash" from the kernel cmdline to see more boot messages. Up to the screenshot everything seems fine, so maybe something breaks networking afterwards.

**Reply by ThierryM** [reply]
Now, I obtain a prompt but nothing works : when I put the login nothing appends, no password demand.
Nothing on the Netconsole, so I join the picture : 
![img1](https://user-images.githubusercontent.com/1470067/99653132-f15fad00-2a58-11eb-9913-c68195a4b042.jpg)
Now, I can change the console with CTRL+ALT+F2 but it's the same state with login impossible.

It seems that the problem is perhaps with ldm...

**Comment by alkisg** [comment]
I guess you mean "lightdm" instead of "ldm", but no, I don't think the problem is there.
Try to ping the client from the server. You should be able to ping it while it's booting, but can you still ping it when it's at the login prompt? If not, it means it lost networking.

Btw do you have any firewalls etc that could interfere with networking?

**Reply by ThierryM** [reply]
Yes, it's lightdm ;-).
As you guess, the ping from the server to the pi works until the prompt appears.

The server is connected on a network with a proxy for the first network card to access Internet. The 2nd network card 192.168.67.1 is connected directly with the client RPI4.

**Comment by alkisg** [comment]
I see that you installed apache, did you install anything else that could break networking?
For example, netplan, wicd, did you set up a wifi connection etc?

**Reply by ThierryM** [reply]
On the server or on the client ?
I think that is on the client. So I know that wicd is installed.
Sorry for this beginner question, but why the booting works fine in NFS_RW mode and not in LTSP netbooting mode ?

**Comment by alkisg** [comment]
In the chroot. See the client boot screenshot above, e.g. "Starting the Apache HTTP Server" means that you installed Apache in the chroot, which may affect client networking. Or you might have installed ufw either on the server or the client.

Since it works in NFS_RW mode, but not in LTSP mode, maybe something in the image cleanup process reveals the problem.

At this point, it would probably be faster to test the https://ltsp.org/docs/installation/raspbian/ page from scratch, without installing any additional programs like apache in the chroot. Once you have LTSP mode working, THEN try to add your programs, and see which one breaks networking.

**Reply by ThierryM** [reply]
Ok. I'll try and I'll report here the results.
Thanks for your patience.
Regard.

**Reply by ThierryM** [reply]
Hi,
I'm testing with the last Raspberry Pi OS image with desktop. That works.
Info : at the first connection, in NFS_RW mode, in the installation process skip the questions about network settings because the connection works directly. The personnalisation of the dektop GUI, Firefox, the links, ... made in NFS_RW mode are resetted for the normal clients.
At the start, a login is asked and the default user is the same as the server.
Now I've to add applications to see where's the problem.

**Comment by ThierryM** [comment]
Hi,
So, after a conversation with the developpers of Primtux, I've took off the parental control of the Primtux distribution (based on the last Raspberry Pi OS) to build a new image. I can boot but now, the client can't connect to Internet In NFS_RW mode.
So where I can fix the network parameters for the clients in order the clients can access to Internet via the 192.168.67.1 network of the ltsp server ?
Thanks for your help, regards,

Thierry

**Comment by alkisg** [comment]
Does your LTSP server have 2 network adapters? If yes, read about "NAT=1" in the ltsp.conf man page, https://ltsp.org/man/ltsp.conf/.

**Reply by ThierryM** [reply]
Yes, I have 2 NIC. So, I've modified the ltsp.conf with NAT=1 in the server section. I've done ltsp kernel image, ltsp initrd and ltsp nfs.
But the problem is the same. When I ping on a terminal, I have this message : "ping: google.com: Name or service unknown".
Under Primtux the network is managed with WICD.

**Comment by alkisg** [comment]
Did you reboot the server or restart the ltsp service?

**Reply by ThierryM** [reply]
Yes. I rebooted the server.

**Comment by alkisg** [comment]
Does the problem happen with the normal Raspberry Pi OS image?

**Reply by ThierryM** [reply]
No.

**Reply by ThierryM** [reply]
But with the parental control on Primtux (my first image), I haven't problem in NFS_RW mode too.

**Comment by alkisg** [comment]
OK then please ask the distribution developers about it, they probably have some weird configuration unrelated to LTSP.

**Reply by ThierryM** [reply]
I've taken off wicd but same problem.
ping 8.8.8.8 works but not ping google.com.

**Comment by alkisg** [comment]
What is the output of these commands on the LTSP server?

```shell
grep dns-server /etc/dnsmasq.d/ltsp-dnsmasq.conf
grep -wo '3500007F:0035' /proc/net/tcp
```

**Reply by ThierryM** [reply]
Hi,
For the command "grep dns-server /etc/dnsmasq.d/ltsp-dnsmasq.conf" : 
`dhcp-option=option:dns-server,172.29.170.152,172.29.170.81`
No problem, it's mine on my network.

For the command "grep -wo '3500007F:0035' /proc/net/tcp" : 
`3500007F:0035`

But, asking with one Primtux developper, there's a script for generating an /etc/resolv.conf file automaticaly at the booting and it fixes "nameserver 127.0.0.1". So I think that the problem is here; as you told. Not with Ltsp.
I dig this.
Thanks you very much.

**Reply by ThierryM** [reply]
The "problem" of the Primtux distribution is that the file /etc/resolv.conf is immunized by the command `chattr +i /etc/resolv.conf`. So I've fixed in that file `nameserver 192.168.67.1` before building the ltsp image.  But, I saw that in nfs, chattr doesn't work so when the client boots, the /etc/resolv.conf is again `nameserver 127.0.0.1`.

**Comment by ThierryM** [comment]
Hi,
I've given up my project to use Primtux distribution for fat clients : too much work to adapt this distribution.
I've advanced therefore and I could use what I've learned with this experience to use Raspberry Pi 4 as clients. I've described this experience (in french) on this page : https://lofurol.fr/joomla/electronique/raspberry-pi/263-utiliser-des-raspberry-pi-4-en-tant-que-clients-lourds. It could help those who would try to use Raspberry Pi 4 as fat clients.
Thank you very much for your help.
Regards,

Thierry
